### PR TITLE
Optimise mysqldump by compressing on server then downloading

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: ruby/data.rb [hunk 1]
```diff
@@ -6,28 +6,23 @@
...
+		Example command: cap -f '/sites/deploynaut/www/assets/Capfile' project1:env1 data:getdb -s data_path=/tmp/mydatabase.sql.gz
 
 		Required arguments to the cap command:
-		data_path - Absolute path (including filename) of where the exported database should be placed, e.g. /tmp/my_database.sql
+		data_path - Absolute path (including filename) of where the exported database should be placed, e.g. /tmp/my_database.sql.gz
 	DESC
 	task :getdb do
-		# todo: output to gzip and stream that back instead of the raw data.
-		run "mysqldump --skip-opt --add-drop-table --extended-insert --create-options --quick --set-charset --default-character-set=utf8 #{mysql_options} -p", :once => true do |channel, stream, data|
+		tmpfile = "/tmp/temp-download-database.sql.gz"
```
Comment (stojg) [inline ruby/data.rb]: Do you think we should future proof it by creating a temporary race condition safe name like http://stackoverflow.com/questions/13787746/creating-a-thread-safe-temporary-file-name ?

File: code/model/DNDataArchive.php [hunk 2]
```diff
@@ -510,15 +510,13 @@ public function fixArchivePermissions($workingDir) {
 	public function setArchiveFromFiles($workingDir) {
 		$commands = array();
 		if($this->Mode == 'db') {
-			$commands[] = 'gzip database.sql';
```
Comment (stojg) [inline code/model/DNDataArchive.php]: the dump is already compressed?

File: ruby/data.rb (hunk 1 above)
Comment (halkyon) [inline ruby/data.rb]: good idea :)

Comment (halkyon) [inline ruby/data.rb]: fixed!

**Pull request merged**